owncloud/core
# Move away from OCA.files.getMimeIcon and use OC.MimeType.getIconUrl

Since we now have the fancy OC.MimeType class (#16724) we should actively use it!

This moves the file app away from the getMimeIcon function that did a request for every mimetype to the server

Testing is easy!
1. Add some files for which we have mimetypes
2. Reload files app and observe the console.. for each mimetype you should see at least 1 request to the server (ajax/mimeicon.php)
3. Now checkout this awesome branch
4. Reload files app
5. Observe no calls to the server for each mimetype (but you still see them)

TODO:
- [x] Remove ajax/mimeicon.php (and its routes)
- [x] Migrate calls in core app to getMimeIcon to OC.MimeType.getIconUrl
